pandas/core/indexes/base.py

```diff
@@ -2378,12 +2378,58 @@ def identical(self, other):
 
     def asof(self, label):
         """
-        For a sorted index, return the most recent label up to and including
-        the passed label. Return NaN if not found.
+        Return the label from the index, or, if not present, the previous one.
 
-        See also
+        Assuming that the index is sorted, return the passed index label if it
+        is in the index, or return the previous index label if the passed one
+        is not in the index.
+
+        Parameters
+        ----------
+        label : object
+            The label up to which the method returns the latest index label.
+
+        Returns
+        -------
+        object : The passed label if it is in the index. The previous label
+            if the passed label is not in the sorted index, or NaN if there
+            is no such label.
```

Comment:
Looks great, just one minor formatting thing. The description goes to the next line, and the colon is not required. Also you can add backticks around the `NaN`. For the rest is perfect, really like this final result.